### PR TITLE
Add configurable year menu 

### PR DESCRIPTION
**Pull request by roine**
Add the year menu like in jquery UI datepicker:

http://api.jqueryui.com/datepicker/#option-changeYear
http://api.jqueryui.com/datepicker/#option-yearRange

### CODE REVIEW

**Comment by bbqbaron** [comment]
Great idea! Playing around with it now and will leave notes if any come up.

File: src/DatePicker/Date.elm [hunk 1]
```diff
@@ -1,6 +1,11 @@
 module DatePicker.Date
     exposing
-        ( initDate
+        ( YearRange
+        , yearRangeActive
+        , between
```
Comment (bbqbaron) [inline src/DatePicker/Date.elm]: `DatePicker.Date` isn't exposed by this library. Won't that mean that no one can actually override `changeYear` in practice? If so, I'm not sure we should expose all of these exports just to get `YearRange` constructors out there. What do you think about re-exporting them from `DatePicker`?

(Personally, I have a bias toward over-exposing API elements, since Elm makes major version upgrades so aggressively anyway, but I inherited the repo and am trying to respect its design thus far.)

File: src/DatePicker.elm [hunk 2]
```diff
@@ -412,9 +419,24 @@ datePicker { today, currentMonth, currentDates, pickedDate, settings } =
         onPicker ev =
             Json.succeed
                 >> onWithOptions ev
-                    { preventDefault = True
+                    { preventDefault = False
```
Comment (bbqbaron) [inline src/DatePicker.elm]: Why this change? `onPicker` seems to be attached to the whole picker, so presumably it has some effect outside the year picker itself?

File: src/DatePicker.elm [hunk 3]
```diff
@@ -412,9 +419,24 @@ datePicker { today, currentMonth, currentDates, pickedDate, settings } =
...
                     , stopPropagation = True
                     }
+
+        onChange handler =
+            on "change" <| Json.map handler <| Json.at [ "target", "value" ] Json.string
```
Comment (bbqbaron) [inline src/DatePicker.elm]: Does `Html.Events.targetValue` not work here?

File: src/DatePicker.elm [hunk 4]
```diff
@@ -412,9 +419,24 @@ datePicker { today, currentMonth, currentDates, pickedDate, settings } =
...
+            year currentMonth == selectedYear
+
+        durationOption selectedYear =
+            ( toString selectedYear
+            , option [ value (toString selectedYear), selected (isCurrentYear selectedYear) ]
+                [ text (toString <| selectedYear) ]
+            )
+
+        dropdownYear =
+            Html.Keyed.node "select" [ onChange ChangeYear, class "year-list" ] (List.map durationOption (yearRange currentMonth settings.changeYear))
```
Comment (bbqbaron) [inline src/DatePicker.elm]: I'm getting a peculiar behavior locally. I don't know if it's `Keyed` related, since I haven't used it before, but it seems like it might be.

Here are some repro steps; can you get this locally?
1. Create an example using `MoreOrLess 5`
2. Open picker
3. Select the earliest possible year (2012)
4. The select now shows "2016".

What's odd is that the debugger shows what looks like correct state behavior: 2012 is the year of `currentMonth`. Does this behavior look familiar? Hoping that you know about any gotchas with `Keyed`, or that I'm barking up the wrong tree entirely :).

File: src/DatePicker.elm (hunk 2 above)
Comment (roine) [inline src/DatePicker.elm]: If you revert this part, the dropdown won't drop on click.

**Comment by roine** [comment]
I moved the yearRange helpers to the Datepicker file but it feels odd. What about having a Datepicker.SettingsHelper which would expose the between, off and moreOrless?

**Comment by bbqbaron** [comment]
@roine Sorry for the delay; busy week. Thanks for fixing the keyed-select issue. I don't mind having the year helpers in the core API: If the user needs to supply them to make the settings work, then it's reasonable to have them there. Maybe at a certain level of complication, we can split the files up. This looks good in local testing.

I have one leftover question, incoming shortly. I'm also happy to add docs for using the option, which I can do after merge.

**Comment by bbqbaron** [comment]
Oh, nope; answered my own question by rereading the code! Great; merging.
